File: src/jvolpecastro/IStrategist.java
package jvolpecastro;

import robocode.JuniorRobot;

public interface IStrategist {
    public abstract void run(JuniorRobot robot);
    public abstract void onScannedRobot(JuniorRobot robot);
    public abstract void onHitByBullet(JuniorRobot robot);
    public abstract void onHitRobot(JuniorRobot robot);
    public void onHitWall(JuniorRobot robot);
}

File: src/jvolpecastro/IStrategist.kt
package jvolpecastro

import robocode.JuniorRobot

interface IStrategist{
    abstract fun run (robot: JuniorRobot)
    abstract fun onScannedRobot(robot: JuniorRobot)
    abstract fun onHitByBullet(robot: JuniorRobot)
    abstract fun onHitRobot(robot: JuniorRobot)
    abstract fun onHitWall(robot: JuniorRobot)
}